packages/zipkin/index.d.ts

```diff
@@ -74,42 +74,29 @@ declare namespace zipkin {
   const randomTraceId: () => string;
 
   namespace option {
-    interface IOption<T> {
-      type: 'Some' | 'None';
-      present: boolean;
-
+    abstract class Option<T> {
       map<V>(fn: (value: T) => V): IOption<V>;
-      ifPresent<V>(fn: (value: T) => V): IOption<V>;
-      flatMap<V>(fn: (value: T) => V): IOption<V>;
-      getOrElse<V>(fnOrValue: (() => V) | V): T;
+      ifPresent(fn: (value: T) => any): void;
+      flatMap<V>(fn: (value: T) => IOption<V>): IOption<V>;
+      getOrElse(fnOrValue: (() => T) | T): T;
       equals(other: IOption<T>): boolean;
       toString(): string;
     }
 
-    interface INone extends IOption<any> {
-      type: 'None';
-      present: false;
-
-      map<V>(fn: (value: any) => V): INone;
-      flatMap<V>(fn: (value: any) => V): INone;
-      equals(other: IOption<any>): boolean;
-      toString(): string;
+    class Some<T> extends Option<T> {
+      constructor(value: T);
+      readonly type: 'Some';
+      readonly present: true;
```

Comment:
May be the definition
```
readonly type: string;
readonly present: boolean;
```
make senses to be included in Option<T> interface.
If I hold a Some or a None, I know its type and present already because they are concrete objects.
Only when I hold a IOption<T>, I didn't know if it is Some or None or if it is present or not, then those property make sense this case.

On the other hand, it is also fine to me of not including it in IOption<T> because I guess this class is designed for internal use only. (Maybe we should not export it to clients at all ^__^)